p2p.go

```diff
@@ -48,6 +46,7 @@ type (
 
 	// Config enumerates the configs required by a host
 	Config struct {
+		ProtocolName             protocol.ID     `yaml:"dhtProtocol"`
```

Comment:
we'd do our best to have keep the consistency of the parameter name and yaml key. so, yaml:"protocolName"